qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,166 @@
+            estimator: An estimator primitive used for calculating expectation values of
+                ``TimeEvolutionProblem.aux_operators``.
+        """
+        if product_formula is None:
+            product_formula = LieTrotter()
+        self.product_formula = product_formula
+        self._estimator = estimator
+
+    @property
+    def estimator(self) -> BaseEstimator | None:
+        """
+        Returns an estimator.
+        """
+        return self._estimator
+
+    @estimator.setter
+    def estimator(self, estimator: BaseEstimator) -> None:
+        """
+        Sets an estimator.
+        """
+        self._estimator = estimator
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        """
+        Whether computing the expectation value of auxiliary operators is supported.
+
+        Returns:
+            ``True`` if ``aux_operators`` expectations in the ``TimeEvolutionProblem`` can be
+            evaluated, ``False`` otherwise.
+        """
+        return True
+
+    def evolve(self, evolution_problem: TimeEvolutionProblem) -> TimeEvolutionResult:
+        """
+        Evolves a quantum state for a given time using the Trotterization method
+        based on a product formula provided. The result is provided in the form of a quantum
+        circuit. If auxiliary operators are included in the ``evolution_problem``, they are
+        evaluated on an evolved state using an estimator primitive provided.
+
+        .. note::
+            Time-dependent Hamiltonians are not yet supported.
+
+        Args:
+            evolution_problem: Instance defining evolution problem. For the included Hamiltonian,
+                ``Pauli`` or ``PauliSumOp`` are supported by TrotterQRTE.
+
+        Returns:
+            Evolution result that includes an evolved state as a quantum circuit and, optionally,
+            auxiliary operators evaluated for a resulting state on an estimator primitive.
+
+        Raises:
+            ValueError: If ``t_param`` is not set to ``None`` in the ``TimeEvolutionProblem``
+                (feature not currently supported).
+            ValueError: If the ``initial_state`` is not provided in the ``TimeEvolutionProblem``.
+            ValueError: If an unsupported Hamiltonian type is provided.
+        """
+        evolution_problem.validate_params()
+        if evolution_problem.t_param is not None:
+            raise ValueError(
+                "TrotterQRTE does not accept a time dependent Hamiltonian,"
+                "``t_param`` from the ``TimeEvolutionProblem`` should be set to ``None``."
+            )
+
+        if evolution_problem.aux_operators is not None and self.estimator is None:
+            warnings.warn(
+                "The time evolution problem contained ``aux_operators`` but no estimator was "
+                "provided. The algorithm continues without calculating these quantities. "
+            )
```

Comment:
Should we better raise an error? If the user has aux_operators but no Estimator that seems to be a configuration problem on their part - would it be better that this was corrected rather than running a partial configuration.

Reply:
Ok, sounds reasonable.